tests/s25Main/integration/testCheats.cpp

```diff
@@ -3,60 +3,100 @@
 
 #include "Cheats.h"
+#include "GamePlayer.h"
+#include "desktops/dskGameInterface.h"
 #include "worldFixtures/CreateEmptyWorld.h"
 #include "worldFixtures/WorldFixture.h"
+#include <turtle/mock.hpp>
 
 BOOST_AUTO_TEST_SUITE(CheatsTests)
 
 namespace {
-template<unsigned T_numPlayers>
-struct CheatsFixture : WorldFixture<CreateEmptyWorld, T_numPlayers>
+constexpr auto numPlayers = 1;
+constexpr auto worldWidth = 64;
+constexpr auto worldHeight = 64;
+struct CheatsFixture : WorldFixture<CreateEmptyWorld, numPlayers, worldWidth, worldHeight>
 {
-    Cheats cheats{this->world};
+    dskGameInterface gameDesktop{game, nullptr, 0, false};
+    Cheats& cheats = gameDesktop.GI_GetCheats();
+    const GameWorldViewer& viewer = gameDesktop.GetView().GetViewer();
+
+    GamePlayer& getPlayer(unsigned id) { return world.GetPlayer(id); }
+    GamePlayer& p1 = getPlayer(0);
+    const MapPoint p1HQPos = p1.GetHQPos();
 };
-using CheatsFixture1P = CheatsFixture<1>;
-using CheatsFixture2P = CheatsFixture<2>;
 } // namespace
 
-BOOST_FIXTURE_TEST_CASE(CheatsAreAllowed_InSinglePlayer, CheatsFixture1P)
-{
-    BOOST_TEST_REQUIRE(cheats.areCheatsAllowed() == true);
-}
-
-BOOST_FIXTURE_TEST_CASE(CheatsAreNotAllowed_InMultiplayer, CheatsFixture2P)
-{
-    BOOST_TEST_REQUIRE(cheats.areCheatsAllowed() == false);
-}
-
-BOOST_FIXTURE_TEST_CASE(CheatModeIsOffByDefault, CheatsFixture1P)
-{
-    BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == false);
-}
-
-BOOST_FIXTURE_TEST_CASE(CanToggleCheatModeOn, CheatsFixture1P)
+BOOST_FIXTURE_TEST_CASE(CanToggleCheatModeOnAndOffRepeatedly, CheatsFixture)
 {
+    BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == false); // initially off
     cheats.toggleCheatMode();
     BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == true);
-}
-
-BOOST_FIXTURE_TEST_CASE(CanToggleCheatModeOnAndOff, CheatsFixture1P)
-{
+    cheats.toggleCheatMode();
+    BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == false);
     cheats.toggleCheatMode();
     BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == true);
     cheats.toggleCheatMode();
     BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == false);
 }
 
-BOOST_FIXTURE_TEST_CASE(CanToggleCheatModeOnAndOffRepeatedly, CheatsFixture1P)
+namespace {
+MOCK_BASE_CLASS(MockGameInterface, GameInterface)
 {
+    MOCK_METHOD(GI_PlayerDefeated, 1)
+    MOCK_METHOD(GI_UpdateMinimap, 1)
+    MOCK_METHOD(GI_FlagDestroyed, 1)
+    MOCK_METHOD(GI_TreatyOfAllianceChanged, 1)
+    MOCK_METHOD(GI_UpdateMapVisibility, 0)
+    MOCK_METHOD(GI_Winner, 1)
+    MOCK_METHOD(GI_TeamWinner, 1)
+    MOCK_METHOD(GI_StartRoadBuilding, 2)
+    MOCK_METHOD(GI_CancelRoadBuilding, 0)
+    MOCK_METHOD(GI_BuildRoad, 0);
+
+    // can't mock noexcept function in turtle - working around it
+    // also clang-format is getting confused around this code - formatting manually
+    // clang-format off
+    MockGameInterface(Cheats& cheats) : cheats_{cheats} {}
+    Cheats& cheats_;
+    Cheats& GI_GetCheats() noexcept override { return cheats_; }
+    // clang-format on
```

Comment:
My clang-format (I think the one coming with Visual Studio) is getting confused here and keeps formatting the `&` placement and `GI_GetCheats()` definition inconsistently with the rest of the codebase.

Reply:
I know of another place where this happens, IIRC in the `tmpFile` part of the test helpers. What does it do here?

Reply:
Both the & placement and definition not being on the same line get rejected by StyleAndFormatting.

Reply:
I see, this is similar to the other place and fine to disable formatting here.